source/Octopus.Tentacle.Tests.Integration/Support/TentacleBuilder.cs

```diff
@@ -13,13 +14,39 @@
 
 namespace Octopus.Tentacle.Tests.Integration.Support
 {
-    public abstract class TentacleBuilder<T> where T : TentacleBuilder<T>
+    public interface ITentacleBuilder
```

Comment:
We needed to apply the environment variables regardless of what type of tentacle it was.

However, while in this space I found it quite awkward having this abstract class with two subclasses. Even the "With" existing methods look a touch awkward (where they return a `T`, where `T` is the type of the subclass...)

Looking at the 2 subclasses, there really isn't much there.

I wonder if it would make things simpler if we just combined all 3 of these into 1 class (this class), and simply had 2 methods like `WithPollingTentacle` and `WithListeningTentacle` instead.

Do we like this idea? (if so, I'll do a separate PR for that)

Reply:
I'm just resolving these as it is adding extra noise and things to read not directly related to this PR.